commit be7a856f5bd251dc84a72eb56cf011facd10988f
Author:     Alexander Bokovoy <ab@samba.org>
AuthorDate: Thu Sep 13 15:40:31 2012 +0300
Commit:     Alexander Bokovoy <ab@samba.org>
CommitDate: Thu Sep 13 17:36:07 2012 +0200

    s3: make ldapsam-related functions a smbldaphelper subsystem
    
    Since these functions are used in pdb_ldap and idmap_ldap, and
    pdb_ldap might be statically linked to libpdb (default), it is
    better to keep them as separate subsystem to avoid polluting libpdb
    namespace.
    
    This is first step in refactoring libpdb. Right now I cannot move
    these functions into proper libsmbldaphelper as it uses more of
    libpdb-included functions and linking pdb_ldap against libsmbldaphelper
    library would have created a loop if pdb_ldap is included into libpdb.
    
    Autobuild-User(master): Alexander Bokovoy <ab@samba.org>
    Autobuild-Date(master): Thu Sep 13 17:36:07 CEST 2012 on sn-devel-104
---
 source3/passdb/wscript_build   | 4 ++--
 source3/winbindd/wscript_build | 2 +-
 source3/wscript_build          | 5 ++++-
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/source3/passdb/wscript_build b/source3/passdb/wscript_build
index 1dfdfd0..d26afc2 100644
--- a/source3/passdb/wscript_build
+++ b/source3/passdb/wscript_build
@@ -1,7 +1,7 @@
 #!/usr/bin/env python
 
 PDB_TDBSAM_SRC =    'pdb_tdb.c'
-PDB_LDAP_SRC =      'pdb_ldap.c pdb_nds.c pdb_ipa.c pdb_ldap_util.c'
+PDB_LDAP_SRC =      'pdb_ldap.c pdb_nds.c pdb_ipa.c'
 PDB_SMBPASSWD_SRC = 'pdb_smbpasswd.c'
 PDB_WBC_SAM_SRC =   'pdb_wbc_sam.c'
 
@@ -15,7 +15,7 @@ bld.SAMBA3_MODULE('pdb_tdbsam',
 
 bld.SAMBA3_MODULE('pdb_ldap',
                  subsystem='pdb',
-                 deps='smbldap',
+                 deps='smbldap smbldaphelper',
                  source=PDB_LDAP_SRC,
                  init_function='',
                  internal_module=bld.SAMBA3_IS_STATIC_MODULE('pdb_ldap'),
diff --git a/source3/winbindd/wscript_build b/source3/winbindd/wscript_build
index 9f11aff..af3741b 100644
--- a/source3/winbindd/wscript_build
+++ b/source3/winbindd/wscript_build
@@ -62,7 +62,7 @@ bld.SAMBA3_MODULE('idmap_passdb',
 bld.SAMBA3_MODULE('idmap_ldap',
                  subsystem='idmap',
                  source=IDMAP_LDAP_SRC,
-                 deps='smbldap',
+                 deps='smbldap smbldaphelper',
                  init_function='',
                  internal_module=bld.SAMBA3_IS_STATIC_MODULE('idmap_ldap'),
                  enabled=bld.SAMBA3_IS_ENABLED_MODULE('idmap_ldap') and bld.env.HAVE_LDAP)
diff --git a/source3/wscript_build b/source3/wscript_build
index c7395e7..c534ae6 100755
--- a/source3/wscript_build
+++ b/source3/wscript_build
@@ -215,7 +215,6 @@ PASSDB_SRC = '''${PASSDB_GET_SET_SRC} passdb/passdb.c
                 passdb/account_pol.c ${PRIVILEGES_SRC}
                 lib/util_nscd.c lib/winbind_util.c
                 passdb/pdb_util.c passdb/pdb_interface.c
-                passdb/pdb_ldap_schema.c
                 passdb/pdb_secrets.c
                 passdb/pdb_unixid.c'''
 #FIXME: lib/winbind_util.c probably is not part of PASSDB_SRC
@@ -733,6 +732,10 @@ bld.SAMBA3_LIBRARY('pdb',
                    vnum='0',
                    vars=locals())
 
+bld.SAMBA3_SUBSYSTEM('smbldaphelper',
+                   source='passdb/pdb_ldap_schema.c passdb/pdb_ldap_util.c',
+                   deps='smbldap secrets3 pdb')
+
 bld.SAMBA3_SUBSYSTEM('SERVER_MUTEX',
                      source=SERVER_MUTEX_SRC,
 		     deps='talloc')
